### PR TITLE
Xan/transcript overlays

### PR DESCRIPTION
**Pull request by XanthosXanthopoulos**
This PR adds support for point cloud overlays over tiled images. The point cloud is rendered on top of images and geometry overlays and maintains the Z values. Both flat coloring as well as categorical rendering are supported

### CODE REVIEW

File: packages/components/src/PointPanel.component.svelte [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+  };
+
+  function datasetOnChange(index: number) {
+    state.selectedDataset = index;
+    state = { ...state };
+  }
+
+  function pointShapeOnChange(index: number) {
+    state.options[state.selectedDataset].pointShape = index;
+    state = { ...state };
```
Comment (SarantopoulosKon) [inline packages/components/src/PointPanel.component.svelte:76]: Let's use `clone` utility now that we have introduced it.

File: packages/components/src/PointPanel.component.svelte [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+            value: state.options[state.selectedDataset].pointShape
+          }
+        }
+      })
+    );
+  }
+
+  function pointSizeOnChange(size: number) {
+    state.options[state.selectedDataset].pointSize = size;
+    state = { ...state };
```
Comment (SarantopoulosKon) [inline packages/components/src/PointPanel.component.svelte:93]: We can use `clone` as well

File: packages/components/src/PointPanel.component.svelte [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+    const feature = features[state.selectedDataset][index];
+
+    if (feature.type === 2) {
+      state.options[state.selectedDataset].selectedAttribute = attributes[
+        state.selectedDataset
+      ].find(x => x.name === feature.attributes[0]);
+    }
+
+    state.options[state.selectedDataset].selectedFeature = index;
+    state = { ...state };
```
Comment (SarantopoulosKon) [inline packages/components/src/PointPanel.component.svelte:152]: Let's use `clone`.

File: packages/core/src/tile/model/point/point.ts [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            const state = updateOptions.groupState?.get(
+              updateOptions.feature.name
+            );
+            const categories = this.meshData[
+              updateOptions.feature.name
+            ] as Int32Array;
+
+            if (state) {
+              for (let index = 0; index < this.groups.length; ++index) {
+                this.groups[index] = state.get(categories[index]) ?? 32;
```
Comment (SarantopoulosKon) [inline packages/core/src/tile/model/point/point.ts:103]: ΝΙΤ: Personally i'd save **32** to a variable, so the variable name can give a hint what 32 is.

File: packages/core/src/tile/worker/loaders/pointLoader.ts [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+        >
+      );
+    } else {
+      if (feature.interleaved) {
+        // All attributes should have the same size
+        // TODO: Add explicit check
+        const elementCount = arrays[feature.attributes[0]].length;
+        const attCount = feature.attributes.length;
+
+        // I use an typed array to work on a continuous block of memory
```
Comment (SarantopoulosKon) [inline packages/core/src/tile/worker/loaders/pointLoader.ts:192]: NIT: 
```diff
- I use an typed array
+ I use a typed array
```

File: packages/core/src/tile/worker/loaders/utils.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+import { getQueryDataFromCache } from '../../../utils/cache';
+import { TypedArray } from '../../types';
+
+export function transformBufferToInt64(
+  buffer: ArrayBuffer,
+  attribute: Attribute
+): BigInt64Array {
+  switch (attribute.type) {
+    case Datatype.Int32:
+      return BigInt64Array.from(
```
Comment (SarantopoulosKon) [inline packages/core/src/tile/worker/loaders/utils.ts:12]: Is there any reason we convert `Int32` to bigInt?

File: packages/core/src/tile/worker/loaders/utils.ts [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  index: string,
+  features: string[]
+): Promise<{ [attribute: string]: TypedArray } | undefined> {
+  const cachedArrays = await Promise.all(
+    features.map(feature => {
+      return getQueryDataFromCache(arrayID, `${feature}_${index}`);
+    })
+  );
+
+  if (cachedArrays.filter(x => x === undefined).length) {
```
Comment (SarantopoulosKon) [inline packages/core/src/tile/worker/loaders/utils.ts:90]: If i understand correctly, we want to return if `cachedArrays` contains an `undefined` value. In that case we can use `Array.prototype.some` which will stop the iteration when the predicate function returns true.

```diff
- if (cachedArrays.filter(x => x === undefined).length) {
+ if (cachedArrays.some(x => x === undefined)) {
```

**Review by SarantopoulosKon — CHANGES_REQUESTED** [review]

File: packages/core/src/tile/worker/loaders/utils.ts (hunk 6 above)
Comment (XanthosXanthopoulos) [reply packages/core/src/tile/worker/loaders/utils.ts:12]: I need all ids to be int64 because that's how the renderer is setup to support picking

**Review by SarantopoulosKon — APPROVED** [review]

**Pull request merged**